registry/registry.go

```diff
@@ -211,8 +212,14 @@ func ContinueOnError(err error) bool {
 		return ContinueOnError(v.Err)
 	case errcode.Error:
 		return shouldV2Fallback(v)
+	case *client.UnexpectedHTTPResponseError:
+		return true
 	}
-	return false
+	// let's be nice and fallback if the error is a completely
+	// unexpected one.
+	// If new errors have to be handled in some way, please
+	// add them to the switch above.
+	return true
```

Comment:
github.com/docker/distribution/issues :smiley_cat: 

In all seriousness, fallback should only happen at the start. Once the process has begun, we should not fallback midway. We've ended up with some odd v1/v2 crossover images in those cases.

This might be one of those, just wait longer and v1 will be gone, so this doesn't matter.